subsys/bluetooth/host/id.c

```diff
@@ -1840,20 +1840,22 @@ int bt_id_set_scan_own_addr(bool active_scan, uint8_t *own_addr_type)
 		 * (through Kconfig).
 		 * Use same RPA as legacy advertiser if advertising.
 		 */
-		if (!IS_ENABLED(CONFIG_BT_SCAN_WITH_IDENTITY) &&
-		    !is_adv_using_rand_addr()) {
-			err = bt_id_set_private_addr(BT_ID_DEFAULT);
-			if (err) {
-				if (active_scan || !is_adv_using_rand_addr()) {
+		if (!IS_ENABLED(CONFIG_BT_SCAN_WITH_IDENTITY)) {
+			if (is_adv_using_rand_addr()) {
+				if (bt_dev.id_addr[BT_ID_DEFAULT].type == BT_ADDR_LE_RANDOM) {
+					*own_addr_type = BT_HCI_OWN_ADDR_RANDOM;
+				} else {
+					*own_addr_type = BT_HCI_OWN_ADDR_PUBLIC;
```

Comment:
Hmmm... with `is_adv_using_rand_addr()` == `true`  .... we have `*own_addr_type = BT_HCI_OWN_ADDR_PUBLIC;` ?

Reply:
is it not possible for the scanner to use the public device address while the advertiser uses the random device address?

`is_adv_using_rand_addr()` also only checks `BT_ADV_ENABLED`